posthog/hogql/printer.py

```diff
@@ -454,6 +461,18 @@ def visit_property_ref(self, ref: ast.PropertyRef):
 
         return property_sql
 
+    def visit_sample_expr(self, ref: ast.SampleExpr):
+        sample_value = self.visit_ratio_expr(ref.sample_value)
+        offset_clause = ""
+        if ref.offset_value:
+            offset_value = self.visit_ratio_expr(ref.offset_value)
+            offset_clause = f" OFFSET {offset_value}"
+
+        return f"SAMPLE {sample_value}{offset_clause}"
+
+    def visit_ratio_expr(self, ref: ast.RatioExpr):
+        return ref.left.value if ref.right is None else f"{ref.left.value}/{ref.right.value}"
```

Comment:
That's dangerous. You never know what's inside `ref.left.value`, so printing it like this is a bad idea. 

We're parsing it now and are sure that `left` is a `Constant` that contains a number... but someone somehow somewhere could change the code in such a way to accidentally expose a hole.

Solution:

```suggestion
        return self.visit(ref.left) if ref.right is None else f"{self.visit(ref.left)}/{self.visit(ref.right)}"
```

This will escape the parts in quotes if they're strings.

Reply:
makes total sense, thanks!